// @dcl/hooks/cjs/hooks/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createAsyncMemoState = exports.useAsyncMemo = exports.useAsyncEffect = exports.usePatchState = exports.useAsyncTasks = exports.useAsyncTask = exports.createAsyncStateState = exports.useAsyncState = exports.useAdvancedUserAgentData = void 0;
// Hooks
var useAdvancedUserAgentData_1 = require("./useAdvancedUserAgentData");
Object.defineProperty(exports, "useAdvancedUserAgentData", { enumerable: true, get: function () { return useAdvancedUserAgentData_1.useAdvancedUserAgentData; } });
var useAsyncState_1 = require("./useAsyncState");
Object.defineProperty(exports, "useAsyncState", { enumerable: true, get: function () { return useAsyncState_1.useAsyncState; } });
Object.defineProperty(exports, "createAsyncStateState", { enumerable: true, get: function () { return useAsyncState_1.createAsyncStateState; } });
var useAsyncTask_1 = require("./useAsyncTask");
Object.defineProperty(exports, "useAsyncTask", { enumerable: true, get: function () { return useAsyncTask_1.useAsyncTask; } });
var useAsyncTasks_1 = require("./useAsyncTasks");
Object.defineProperty(exports, "useAsyncTasks", { enumerable: true, get: function () { return useAsyncTasks_1.useAsyncTasks; } });
var usePatchState_1 = require("./usePatchState");
Object.defineProperty(exports, "usePatchState", { enumerable: true, get: function () { return usePatchState_1.usePatchState; } });
var useAsyncEffect_1 = require("./useAsyncEffect");
Object.defineProperty(exports, "useAsyncEffect", { enumerable: true, get: function () { return useAsyncEffect_1.useAsyncEffect; } });
var useAsyncMemo_1 = require("./useAsyncMemo");
Object.defineProperty(exports, "useAsyncMemo", { enumerable: true, get: function () { return useAsyncMemo_1.useAsyncMemo; } });
Object.defineProperty(exports, "createAsyncMemoState", { enumerable: true, get: function () { return useAsyncMemo_1.createAsyncMemoState; } });
