// decentraland-ui2/dist/components/Address/Address.stories.js
import { jsx as _jsx } from "react/jsx-runtime";
import { Address } from "./";
const address = "0x0F5D2fB29fb7d3CFeE444a200298f468908cC942";
const meta = {
    component: Address,
    title: "Decentraland UI/Address",
    tags: ["autodocs"],
    argTypes: {
        value: {
            description: "The Ethereum address to display (required)",
            defaultValue: "",
        },
        strong: {
            description: "Bold the address text",
            defaultValue: false,
        },
        shorten: {
            description: "Shorten the address for display",
            defaultValue: true,
        },
        tooltip: {
            description: "Show full address on hover",
        },
    },
    render: (args) => _jsx(Address, { ...args }),
};
const FullAddress = {
    name: "Full Address",
    args: {
        value: address,
        shorten: false,
    },
};
const ShortAddress = {
    name: "Short Address",
    args: {
        value: address,
        shorten: true,
    },
};
const StrongStyleAddress = {
    name: "Strong Style Address",
    args: {
        value: address,
        strong: true,
    },
};
const WithTooltip = {
    name: "With Tooltip",
    args: {
        value: address,
        tooltip: true,
    },
};
// eslint-disable-next-line import/no-default-export
export default meta;
export { FullAddress, ShortAddress, StrongStyleAddress, WithTooltip };
