// @dcl/schemas/dist/misc/catalyst-deployment.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CatalystDeployment = void 0;
const validation_1 = require("../validation");
const auth_chain_1 = require("./auth-chain");
var CatalystDeployment;
(function (CatalystDeployment) {
    CatalystDeployment.schema = {
        type: 'object',
        required: ['entity'],
        properties: {
            entity: {
                type: 'object',
                required: ['entityId', 'entityType', 'authChain'],
                properties: {
                    entityId: { type: 'string' },
                    entityType: { type: 'string' },
                    authChain: auth_chain_1.AuthChain.schema,
                    metadata: { type: 'object', additionalProperties: true, nullable: true }
                },
                additionalProperties: true
            }
        },
        additionalProperties: true
    };
    CatalystDeployment.validate = (0, validation_1.generateLazyValidator)(CatalystDeployment.schema);
})(CatalystDeployment = exports.CatalystDeployment || (exports.CatalystDeployment = {}));
