// decentraland-ui2/dist/components/AvatarFace/AvatarFace.stories.js
import { jsx as _jsx, Fragment as _Fragment, jsxs as _jsxs } from "react/jsx-runtime";
import { Typography } from "@mui/material";
import { AvatarFace } from "./AvatarFace";
import { avatar } from "../../data/avatar";
import { Mana } from "../Mana";
const meta = {
    component: AvatarFace,
    title: "Decentraland UI/AvatarFace",
    tags: ["autodocs"],
    argTypes: {
        size: {
            description: "Size of the avatar face",
        },
        inline: {
            description: "Avatar component should be displayed inline",
        },
    },
    render: (args) => _jsx(AvatarFace, { ...args }),
};
const Guest = {
    args: {},
};
const Simple = {
    args: {
        size: "large",
        avatar,
    },
};
const Sizes = {
    args: {
        avatar,
    },
    render: (args) => (_jsxs(_Fragment, { children: [_jsx(AvatarFace, { size: "tiny", ...args }), _jsx(AvatarFace, { size: "small", ...args }), _jsx(AvatarFace, { size: "medium", ...args }), _jsx(AvatarFace, { size: "large", ...args }), _jsx(AvatarFace, { size: "responsive", ...args })] })),
};
const InParagraph = {
    name: "In a paragraph",
    args: {
        avatar,
        inline: true,
        size: "tiny",
    },
    render: (args) => (_jsxs(Typography, { children: ["You've transferred ", _jsx(Mana, { inline: true, children: "1,000" }), " to", " ", _jsx(AvatarFace, { ...args }), " ", _jsx("strong", { children: "cazala" })] })),
};
// eslint-disable-next-line import/no-default-export
export default meta;
export { Guest, Simple, Sizes, InParagraph };
