// decentraland-ui2/dist/components/UserMenu/UserMenuSignedIn/UserMenuActions.stories.js
import { jsx as _jsx } from "react/jsx-runtime";
import { UserMenuActions } from "./UserMenuSignedIn";
import { avatar } from "../../../data/avatar";
import { i18n } from "../UserMenu.i18n";
const meta = {
    component: UserMenuActions,
    title: "Decentraland UI/User Menu Actions",
    tags: ["autodocs"],
    argTypes: {},
    args: {
        onClickMarketplaceAuthorization: () => console.log,
        onClickMyAssets: () => console.log,
        onClickProfile: () => console.log,
        onClickSignOut: () => console.log,
        onClickAccount: () => console.log,
        trackingId: "",
        i18n: i18n,
    },
    render: (args) => _jsx(UserMenuActions, { ...args }),
};
const Sample = {
    args: {
        avatar: avatar,
    },
};
const Guest = {
    args: {
        avatar: { ...avatar, hasClaimedName: false },
    },
};
const WithoutProfile = {
    args: {
        address: avatar.ethAddress,
    },
};
// eslint-disable-next-line import/no-default-export
export default meta;
export { Sample, Guest, WithoutProfile };
