// decentraland-ui2/dist/components/DecentralandFont/FontFaceSelector.stories.js
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import { Box, Typography } from "@mui/material";
const meta = {
    component: Typography,
    title: "Decentraland UI/Decentraland Font",
    tags: ["autodocs"],
    argTypes: {
        variant: {
            name: "Font type",
            description: "The typography variant to use",
            control: {
                type: "select",
            },
            options: [
                "h1",
                "h2",
                "h3",
                "h4",
                "h5",
                "h6",
                "subtitle1",
                "subtitle2",
                "body1",
                "body2",
                "caption",
                "overline",
            ],
        },
    },
    render: (args) => (_jsx(Box, { children: _jsx(Typography, { sx: { fontFamily: "DecentralandHero" }, variant: "h1", ...args, children: "Decentraland" }) })),
};
const Simple = {};
const All = {
    render: () => (_jsxs(Box, { children: [_jsx(Typography, { sx: { fontFamily: "DecentralandHero" }, variant: "h1", children: "Decentraland" }), _jsx(Typography, { sx: { fontFamily: "DecentralandHero" }, variant: "h2", children: "Decentraland" }), _jsx(Typography, { sx: { fontFamily: "DecentralandHero" }, variant: "h3", children: "Decentraland" }), _jsx(Typography, { sx: { fontFamily: "DecentralandHero" }, variant: "h4", children: "Decentraland" }), _jsx(Typography, { sx: { fontFamily: "DecentralandHero" }, variant: "h5", children: "Decentraland" }), _jsx(Typography, { sx: { fontFamily: "DecentralandHero" }, variant: "h6", children: "Decentraland" }), _jsx(Typography, { sx: { fontFamily: "DecentralandHero" }, variant: "subtitle1", children: "Decentraland" }), _jsx(Typography, { sx: { fontFamily: "DecentralandHero" }, variant: "subtitle2", children: "Decentraland" }), _jsx(Typography, { sx: { fontFamily: "DecentralandHero" }, variant: "body1", children: "Decentraland" }), _jsx(Typography, { sx: { fontFamily: "DecentralandHero" }, variant: "body2", children: "Decentraland" }), _jsx(Typography, { sx: { fontFamily: "DecentralandHero" }, variant: "caption", children: "Decentraland" }), _jsx(Typography, { sx: { fontFamily: "DecentralandHero" }, variant: "overline", children: "Decentraland" })] })),
};
// eslint-disable-next-line import/no-default-export
export default meta;
export { Simple, All };
